File: src/main/java/com/gitlab/daring/image/util/EnumUtils.java
package com.gitlab.daring.image.util;


public class EnumUtils {

    public static <T> T findEnum(T[] vs, String name) {
        String ln = name.toLowerCase();
        for (T v : vs) {
            String vn = v.toString().toLowerCase();
            if (vn.startsWith(ln)) return v;
        }
        throw new IllegalArgumentException("name=" + name);
    }

    public static <T extends Enum<T>> T findEnum(Class<T> cl, String name) {
        return findEnum(cl.getEnumConstants(), name);
    }

    private EnumUtils() {
    }

}

File: src/main/kotlin/com/gitlab/daring/image/util/EnumUtils.kt
package com.gitlab.daring.image.util


object EnumUtils {

    @JvmStatic
    fun <T> findEnum(vs: Array<T>, name: String): T {
        val ln = name.toLowerCase()
        for (v in vs) {
            if ("$v".toLowerCase().startsWith(ln)) return v
        }
        throw IllegalArgumentException("name=" + name)
    }

    @JvmStatic
    fun <T : Enum<T>> findEnum(cl: Class<T>, name: String): T {
        return findEnum(cl.enumConstants, name)
    }

}
